yarnitor/background/worker.py

```diff
@@ -320,12 +325,23 @@ def run_task(app):
             if std_info is None:
                 ah = BaseHandler.from_yarn_application_info(app)
                 std_info = ah.generate_standardized_info(app)
-                std_info["state"] = "NON_RESPONSIVE"
+                std_info["state"] = "UNKNOWN"
 
             return std_info
 
-        aresult = threadpool.map(run_task, apps)
-        result = {info["id"]: info for info in list(aresult)}
+        # Wait for all async results, raise if it takes too long
+        async_result = threadpool.map(run_task, apps, timeout=THREADPOOL_TIMEOUT)
+        # Materialize results as a list, we need them all anyway
+        results = list(async_result)
+
+        # If all of the applications are non-responsive, then it's quite possible
+        # the YARN proxy is down and the true state of everything should be unknown
+        if unknown_state == len(results):
```

Comment:
is this saying "if 0 == len(results)"? I don't see where unknown_state is ever increased from 0